File: app/src/test/java/com/benoitquenaudon/tvfoot/red/util/InjectionContainer.java
package com.benoitquenaudon.tvfoot.red.util;

import com.benoitquenaudon.tvfoot.red.app.injection.component.TestComponent;

public final class InjectionContainer {
  private TestComponent testComponentInstance;

  public TestComponent testComponent() {
    if (testComponentInstance == null) {
      testComponentInstance = Dagger2Helper.buildComponent(TestComponent.class);
    }
    return testComponentInstance;
  }
}

File: app/src/test/java/com/benoitquenaudon/tvfoot/red/testutil/InjectionContainer.kt
package com.benoitquenaudon.tvfoot.red.testutil

import com.benoitquenaudon.tvfoot.red.app.injection.component.TestComponent

object InjectionContainer {
  val testComponentInstance: TestComponent by lazy {
    Dagger2Helper.buildComponent(TestComponent::class.java)
  }
}
